plugins/cache/cache.go

```diff
@@ -79,8 +79,8 @@ func (p *provider) NewPlugin(options plugins.PluginOptions) (plugins.Plugin, err
 		engine:         options.Engine,
 		environment:    options.Environment,
 		monitor:        options.Monitor,
-		sharedCache:    caching.New(constructor, false, options.Environment.GarbageCollector),
-		exclusiveCache: caching.New(constructor, false, options.Environment.GarbageCollector),
+		sharedCache:    caching.New(constructor, false, options.Environment.GarbageCollector, options.Monitor),
```

Comment:
`.WithPrefix('shared-cache')` and simlar for exclusive...

What else are the metric names going to be?

Reply:
so I guess the questions is if we want them aggregated together or not...

We could get names:
```
shared-cache.cache-hit
shared-cache.cache-miss
shared-cache.constructor-duration
exclusive-cache.cache-hit
exclusive-cache.cache-miss
exclusive-cache.constructor-duration
```
or we could get:
```
cache-hit
cache-miss
constructor-duration
```
if we keep it... my question is if other place there are metrics recorded to `c.monitor` without any prefixes... I don't actually mind aggregating them, that seems decent (I think).

Reply:
Can there be instances where you would like to see what should be shared and what should be exclusive for better performance. Otherwise better to keep them aggregated to reduce noise..., you will definitely have better idea.

Reply:
I think reducing noise is useful... depending on how we use caches these stats could easily be somewhat irrelevant anyways.

Yes, it's cool to have them... for other caches they'll make a lot more sense..